src/ESLint.Formatter/sarif.js

```diff
@@ -220,6 +222,15 @@ module.exports = function (results, data) {
                         };
                     }
 
+                    if (message.suppressions) {
```

Comment:
>message.suppressions [](http://example.com/codeflow?start=24&length=20)

An issue to keep in mind: in the case of *any* suppression utilization, SARIF requires that all messages populate this element (with an empty array in cases the result is unsuppressed. Spec text follows.

Many systems only conditionally emit or otherwise process suppression details. The .NET SuppressMessageAttribute is a good example: this in-source data is only produced in the presence of a certain #define at compilation time.

If ESLint *always* provides a suppression mechanism, one choice you have is to always emit an empty array for every result (at the expense of increasing log file size).

Alternately, if there's an efficient way to determine that *no* message in an analysis is suppressed, you can omit this data.,

```
The suppressions values for all result objects in theRun SHALL be either all null or all non-null.
NOTE: The rationale is that an engineering system will generally evaluate all results for suppression, or none of them. Requiring that the suppressions values be either all null or all non-null enables a consumer to determine whether suppression information is available for the run by examining a single result object.
```

Reply:
Fixed. Thanks.